### PR TITLE
add ability to vendor from generic git repos; add proto patcher

### LINKED ISSUE #13: enable vendoring git repos
currently we only support vendoring from go projects specified in go.mod. we should provide a mechanism to vendor any file from any git repo

### LINKED ISSUE #14: enable patching proto files
a common use case is to vendor proto files and then patch them (e.g. set their `go_package` option). anyvendor should make it easy for us to match files on globs and then patch them

### PR DESCRIPTION
**Pull request by ilackarms**
BOT NOTES: 
resolves https://github.com/solo-io/anyvendor/issues/13
resolves https://github.com/solo-io/anyvendor/issues/14

### CODE REVIEW

File: pkg/git/vendor_git_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/anyvendor/pkg/git"
+)
+
+var _ = Describe("VendorGit", func() {
```
Comment (EItanya) [inline pkg/git/vendor_git_test.go:11]: Nit: Can we clean this up in a `AfterEach`

File: pkg/git/cache.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	url, sha, tag, authUser, authToken string,
+) error {
+	repoDir, _ := c.GetRepoDir(url)
+	repoExists, err := fileExists(repoDir)
+	if err != nil {
+		return err
+	}
+	var authMethod transport.AuthMethod
+	if authToken != "" {
+		authMethod = &http.BasicAuth{Username: authUser, Password: authToken}
```
Comment (EItanya) [inline pkg/git/cache.go:43]: How would this work with a GITHUB_TOKEN in CI?

Comment (ilackarms) [reply pkg/git/cache.go:43]:
```go
AuthUser:  os.Getenv("GITHUB_USER"),  
AuthToken:  os.Getenv("GITHUB_TOKEN"),
```

Comment (ilackarms) [reply pkg/git/cache.go:43]: `GITHUB_TOKEN` is all you need

**Review by EItanya — APPROVED** [review]

**Pull request merged**